### DISCUSSION TITLE
App with beeware and toga

### DISCUSSION THREAD

**Discussion by NickVanelli** [beeware/toga#1985] [Q&A]
.

**Comment by freakboy3742** [comment]
In the case of (1) - `briefcase run android -d @myapp -u` is asking briefcase to run your app on Android using the `myapp` emulator image. If you haven't set up a `myapp` emulator, this won't work. To get your app running, leave off the `-d @mapp` argument, and you'll be prompted to create an emulator, or select from existing emulators on your computer (or devices attached to your computer).

In the case of (2) - Unfortunately, ChatGPT has lead you astray here. Despite appearances, ChatGPT isn't magic. It's only as good as the sample code that it is learning from, and unfortunately there isn't enough sample BeeWare code in the wild for it to learn how to use BeeWare APIs correctly.

The second problem is that you've used Plyer in a BeeWare app. Plyer is a Kivy project; as a result, it uses Kivy's mechanisms for accessing system libraries. On desktop, this doesn't matter so much - Kivy's system bridge and Toga's system bridge can mostly cooperate. Your desktop app will be larger than it needs to be, but it should *work*. 

Android is another story. Getting system libraries working on Android (and iOS, for that matter) is a delicate process that requires deep integration with the operating system - and as a result, you can't currently use Plyer on Android or iOS if you want to use Toga and/or BeeWare to deploy your app.

Unfortunately, I don't have an option for you here - Toga doesn't currently have an equivalent of Plyer providing cross-platform API for notifications.
